includes/Registry/Registry.php

```diff
@@ -257,6 +258,41 @@ static function ( $post_type ) {
 		}//end foreach
 	}
 
+	/**
+	 * Registers custom block types.
+	 *
+	 * The 'core/synced-pattern' block is a reusable synced pattern block and
+	 * is not meant to appear in the block editor selection but is used
+	 * internally for resolving reusable content.
+	 */
+	protected function register_custom_block_types(): void {
+		$registry   = \WP_Block_Type_Registry::get_instance();
+		$block_name = 'core/synced-pattern';
+
+		if ( ! $registry->is_registered( $block_name ) ) {
```

Comment:
This seems very fragile and maybe also unnecessary?
Instead I would recommend reusing `CoreBlock` by renaming it in GraphQL to `CoreSyncedPattern`

Reply:
@justlevine How can we rename an existing block in GraphQL? I'm not familiar with this.

Reply:
@theodesp  how should we handle the mapping in GraphQL to reflect core/synced-pattern without re-registering it?

Reply:
https://github.com/wp-graphql/wp-graphql/blob/eeff88cde5ef09095f4718d48cde534c9ba80a55/src/Type/WPObjectType.php#L85

Pseudocode:

```php
add_filter( 'graphql_type_name', static function( $type_name ) {
  if ( 'CoreBlock' !== $type_name && 'CoreBlockAttributes' !== $type_name ) { return $type_name; }
  
  return str_replace( 'CoreBlock', 'CoreSyncedPatternBlock', $type_name );
} );
```

Or you could probably use `includes/Block/CoreBlock.php` to intercept and overload the default registration.